### PR TITLE
Initial Covid Support UI changes AB#10989

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
# Fixes or Implements [AB#10989](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10989)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

New covid support screen and nav menu
image
image

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [x] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES | NO

### Browsers Tested

-   [ ] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

File: Apps/AdminWebClient/src/ClientApp/src/views/CovidCard.vue [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+                    Mail
+                    <v-icon class="ml-2" size="sm">fas fa-paper-plane</v-icon>
+                </v-btn>
+            </v-row>
+        </v-form>
+    </v-container>
+</template>
+
+<style lang="scss">
+.dashboard-rating-card {
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/views/CovidCard.vue:370]: Is this class being used? cant see it above being referenced

File: Apps/AdminWebClient/src/ClientApp/src/services/interfaces.ts [hunk 2]
```diff
@@ -69,6 +70,11 @@ export interface ISupportService {
     ): Promise<MessageVerification[]>;
 }
 
+export interface ICovidSupportService {
+    initialize(http: IHttpDelegate): void;
+    searchByPHN(phn: string): Promise<CovidCardRequestResult>;
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/services/interfaces.ts:75]: On the controller side there are three paths:
Retrieve covid info -> Patient Info + Immz Events
Retrieve Doc -> base 64 doc + file name (similar to webclient)
Mail Doc -> boolean (true if sucessful)

**Comment by tiagograf-AOT** [comment]
Missing the interpretation of multiple  address lines that comes from the backend model, suggestions include using a text area or possibly limiting to 3 lines. PR is up for graps!

**Comment by MikeLyttle** [comment]
image

File: Apps/AdminWebClient/src/ClientApp/src/models/address.ts [hunk 3]
```diff
@@ -1,20 +1,17 @@
 // Model that provides a user representation of addresses.
-export default interface Address {
-    // Gets or sets the address.
-    address: string;
-
-    // Gets or sets the second address line.
-    address2?: string;
+export default class Address {
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/models/address.ts:2]: if this only represents a model that does not provide any methods or calculated fields we use an interface.

File: Apps/AdminWebClient/src/ClientApp/src/models/covidCardDocumentResult.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 export default class CovidCardDocumentResult {
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/models/covidCardDocumentResult.ts:1]: Same here, interface is preferred

File: Apps/AdminWebClient/src/ClientApp/src/models/covidCardPatientResult.ts [hunk 5]
```diff
@@ -3,8 +3,8 @@ import PatientData from "@/models/patientData";
 
 export default class CovidCardPatientResult {
```
Comment (FuriousLlama) [inline Apps/AdminWebClient/src/ClientApp/src/models/covidCardPatientResult.ts:4]: same here as well

**Review by FuriousLlama — COMMENTED** [review]
In TS  we should use classes only for objects that provide functionality. Dummy object representations or POCOs Interfaces are preferred since they only define the type of data

**Review by FuriousLlama — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**